compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -241,10 +241,6 @@ impl Attribute {
 }
 
 impl AttrItem {
-    pub fn span(&self) -> Span {
-        self.args.span().map_or(self.path.span, |args_span| self.path.span.to(args_span))
```

Comment:
Are you sure we can't save this by using one of the span ancestor functions like https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/span_encoding/struct.Span.html#method.find_ancestor_inside?

Reply:
Maybe just remove the span arithmetic and not try to improve spans for macro scenarios?

Reply:
I don't see a way to get `find_ancestor_in_same_ctxt` to work. The issue is that in the case of `#[link_section = $s]`, the span for `$s` is not stored anywhere. The `AttrItem` stores the path as `link_section` (or `$e` in the other example, it is before expansion). The `AttrArgs` stores the `eq_span` for the `=` in the macro, and the value as `"foo"` from the invocation.

I think supporting proper suggestions in macros is pretty important, as this pattern comes up quite frequently. I realize this can only be tuned to support certain scenarios, and not be perfect in all situations. However, I think recording the span as implemented in this PR covers the vast majority of situations well.

Are you still concerned with the regressions here?